### PR TITLE
sys-block/thin-provisioning-tools: bump 1.0.6

### PR DESCRIPTION
**Pull request by Peter-Levine**
- **sys-block/thin-provisioning-tools: remove unsupported archs**
Though it may change with Rust/GCC, Rust/LLVM doesn't support alpha,
hppa, or ia64.  And until https://bugs.gentoo.org/769452 is sorted
out, arm, ppc, s390, and sparc will emit "nonsolvable depset(bdepend)"
error.

-  **sys-block/thin-provisioning-tools: add CRATES for non-live ebuilds**
Add CRATES to make it easier to use the 9999 ebuild as a general
template while only updating the crate versions when needed.

-  **sys-block/thin-provisioning-tools: update LICENSE**
Add licenses for cargo pkgs.  `cargo-license --all-features` shows
Apache-2.0, BSD, GPL-3, MIT, and Unicode-DFS-2016 as hard
requirements. These also happen to satisfy all ORed license
requirements as well.

-  **sys-block/thin-provisioning-tools: add io-uring USE flag**
Add experimental support for io uring.

-  **sys-block/thin-provisioning-tools: remove src_compile**
With emake, the Makefile just runs `cargo build --release` which
doesn't use ECARGO_ARGS set in cargo_src_configure() and doesn't
respect the debug USE flag. Let cargo_src_compile() handle building.

-  **sys-block/thin-provisioning-tools: don't use emake to install binary**
For USE=debug to work, portage needs to run cargo_src_compile(), but
if `emake` is used in src_install(), it will trigger a rebuild without
debug.  If cargo_src_install() is used instead, manpages and symlinks
won't be installed and the binary would have to be moved since there's
no way to make cargo install to /usr/sbin. So remove $(PDATA_TOOLS)
dependency in Makefile instead.  Might as well patch out $(STRIP) too.

-  **sys-block/thin-provisioning-tools: add DOCS**
Add DOCS to include TODO.md and notes.md, and call `einstalldocs` in
src_install().

- **sys-block/thin-provisioning-tools: add 1.0.6**

Signed-off-by: Peter Levine <plevine457@gmail.com>

### CODE REVIEW

File: sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild [hunk 1]
```diff
@@ -27,14 +170,21 @@ src_unpack() {
...
 	else
-		default
+		cargo_src_unpack
 	fi
 }
 
-src_compile() {
-	emake V= STRIP=true
+src_configure() {
+	local myfeatures=( $(usex io-uring io_uring "") )
```
Comment (SoapGentoo) [inline sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild:178]:
```suggestion
	local myfeatures=( $(usev io-uring) )
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild (hunk 1 above)
Comment (Peter-Levine) [reply sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild:178]: The feature name uses an underscore while the USE flag, common to six other packages, uses a hyphen.

Comment (mattst88) [reply sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild:178]: Indeed, and in fact we have a documented policy against using underscores in `USE` flag names outside of `USE_EXPAND`. See https://projects.gentoo.org/qa/policy-guide/use-flags.html#pg0803

Comment (SoapGentoo) [reply sys-block/thin-provisioning-tools/thin-provisioning-tools-9999.ebuild:178]: oh sorry yes, then use `$(usev io-uring io_uring)`

**Comment by thesamesam** [comment]
Thank you Peter! Please consider joining as a co-maintainer if you're using thin-provisioning-tools as well.

**Pull request closed without merging**